File: 6KYU/Take a Ten Minutes Walk/solution.java
public class TenMinWalk {
  public static boolean isValid(char[] walk) {
    int ns = 0; int ew = 0;
    
    for(int i = 0; i < walk.length; i++) {
      if(walk[i] == 'n') {
        ns++;
      } else if(walk[i] == 's') {
        ns--;
      } else if(walk[i] == 'e') {
        ew++;
      }
      if(walk[i] == 'w') {
        ew--;
      }
    }
    return walk.length == 10 && ns == 0 && ew == 0;
  }
}

File: 6KYU/Take a Ten Minutes Walk/solution.go
package kata

func IsValidWalk(walk []rune) bool {
	if len(walk) != 10 {
		return false
	}

	dirs := make(map[rune]int)
	dirs['n'] = 0
	dirs['e'] = 0
	dirs['s'] = 0
	dirs['w'] = 0

	for _, e := range walk {
		dirs[e] = dirs[e] + 1
	}

	return dirs['n'] == dirs['s'] && dirs['e'] == dirs['w']
}
